packages/golden-layout/src/utils/EventUtils.ts

```diff
@@ -1,20 +1,23 @@
 import { useEffect } from 'react';
 import EventEmitter from './EventEmitter';
 
+export type EventHandlerFunction<P = []> = (
+  ...parameters: P extends unknown[] ? P : [P]
+) => void;
```

Comment:
I don't care about changing it since it's only in 2 places, but just something to note for future reference is you could turn the extends logic into a helper type so you don't need to reuse it

```ts
type AsArray<P> = P extends unknown[] ? P : [P];

export type EventHandlerFunction<P = []> = (...parameters: AsArray<P>) => void;
```